### PR TITLE
Faceted plot implementation proposal

### PR DESCRIPTION
**Pull request by bobular**
Here is a `web-components`-only approach to implementing faceting that I would like feedback and suggestions for

Check out the last two stories in PiePlot.stories.tsx to see how faceted data looks (very simple version for now)

### CODE REVIEW

**Comment by bobular** [comment]
Do not be afraid of the failed checks.  Just go to the Pie plot stories to test this out.

I have [looked into plotless plots](https://github.com/VEuPathDB/web-components/issues/130) (with just the legend) and I didn't find anything promising.  We may have to implement our own.  This should be tied to the colour scheme/palette management.

**Comment by bobular** [comment]
> Hey Bob, I'm just reviewing the code (not testing anything). This looks great. I'm not seeing a strong reason for `FacetedPlot` to accept non-faceted data. That would reduce some of the branching code. Thoughts?

I don't know. Don't we want to avoid code like this in the viz components? (Which adds a branch.)

```
  { isFaceted(data)
    ? (<FacetedPlot data={data} component={FooComponent} props={plotProps} />)
    : (<FooComponent data={data} {...plotProps})
  }
```
I guess it's OK if all the plotProps are bundled up into a variable as I've done above.

**Comment by bobular** [comment]
I would like to merge this and the sibling components PR ASAP but this will require some basic wiring of ALL the visualizations the same as I already did for Barplot.  Could take an hour for each.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -24,15 +24,17 @@ A clear and concise description of what you expected to happen.
 If applicable, add screenshots to help explain your problem.
 
 **Desktop (please complete the following information):**
- - OS: [e.g. iOS]
- - Browser [e.g. chrome, safari]
- - Version [e.g. 22]
+
```
Comment (bobular) [inline .github/ISSUE_TEMPLATE/bug_report.md:27]: I didn't make these changes!

Comment (dmfalke) [reply .github/ISSUE_TEMPLATE/bug_report.md:27]: It looks like this branch is behind `main`. I'm sure if you merge the latest, this will go away.

**Review by dmfalke — APPROVED** [review]

**Review by jtlong3rd — APPROVED** [review]

**Pull request merged**